base/src/main/java/com/smartdevicelink/managers/screen/SoftButtonObject.java

```diff
@@ -309,4 +309,24 @@ public void setOnEventListener(OnEventListener onEventListener) {
          */
         void onUpdate();
     }
+
+    /**
+     * @param o
+     * @return true if two SoftButtonObjects have the same name.
+     */
+    @Override
+    public boolean equals(Object o) {
+        if(o == null){
+            return false;
+        }
+        if(o instanceof SoftButtonObject) {
```

Comment:
There is a more standard way to implement the `equals()` method using `hashCode()`. We use that in some classes in the project like `MenuCell` and `ChoiceCell`